src/components/genreSelect/genreSelect.test.tsx

```diff
@@ -0,0 +1,54 @@
+import { describe, expect, test } from 'vitest';
+import { screen, render } from '@testing-library/react';
+import GenreSelect from './genreSelect';
+import user from '@testing-library/user-event';
+import { vi } from 'vitest';
+
+describe('Genre Select tests', () => {
+  test('check component renders all genres passed in props', async () => {
+    user.setup();
+    const SelectGenreMockHandler = vi.fn();
+    const genreList: Genre[] = ['Horror', 'Comedy', 'Romantic'];
+    render(
+      <GenreSelect
+        genreList={genreList}
+        selectedGenre="Comedy"
+        onSelect={SelectGenreMockHandler}
+      />
+    );
+    const genreListItems = screen.getAllByTestId('genreListItem');
+    expect(genreListItems.length).toBe(genreList.length);
+  });
+
+  test('check component highlights the selected genre', async () => {
```

Comment:
1) Why async?
2) user.setup()?  any user interaction in the test
3)The goal of test is to confirm that the selected genre has different appearance compairing with others. This test doesnt make it. You need to confirm that 'selectedGenreItem' has text 'Horror'